### PR TITLE
Var init checker

### PR DESCRIPTION
**Pull request by Elbargho**

### CODE REVIEW

File: src/ex3/Lattice.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	public boolean isInit(String varID) {
+		for (LatticeVar var : set) {
+			if (varID.equals(var.VarID)) {
+				if (var.isInit == initKind.tt)
+					return true;
+				else
+					return false;
+			}
+		}
+		return true;
```
Comment (ronbel) [inline src/ex3/Lattice.java:41]: Shouldn't this be false? Or throw an exception?

File: src/ex3/VariableInitVisitor.java [hunk 2]
```diff
@@ -5,172 +5,218 @@
...
+	@Override
+	public void visit(IfStatement ifStatement) {
+		Lattice thenLat, elseLat;
+		thenLat = new Lattice();
+		elseLat = new Lattice();
+		thenLat.copy(lattice);
+		elseLat.copy(lattice);
+		ifStatement.cond().accept(this);
+		lattice = thenLat;
+		lattice.assign("y");
```
Comment (ronbel) [inline src/ex3/VariableInitVisitor.java]: What does this line do?

File: src/ex3/Lattice.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package ex3;
+
+import java.util.*;
+
+public class Lattice {
+	private List<LatticeVar> set;
```
Comment (ronbel) [inline src/ex3/Lattice.java:6]: Could have been easier with a hashmap

File: src/ex3/VariableInitVisitor.java [hunk 4]
```diff
@@ -5,172 +5,218 @@
...
+	}
+
+	@Override
+	public void visit(ArrayLengthExpr e) {
 
-    private final ClassMapping classMap;
+	}
+
+	@Override
+	public void visit(MethodCallExpr e) {
```
Comment (ronbel) [inline src/ex3/VariableInitVisitor.java]: Should probably check that all args have been initialized

File: src/ex3/VariableInitVisitor.java (hunk 2 above)
Comment (Elbargho) [reply src/ex3/VariableInitVisitor.java]: test mode

File: src/ex3/Lattice.java (hunk 1 above)
Comment (Elbargho) [reply src/ex3/Lattice.java:41]: at this specific line "41" I reach the end of for loop so I didn't find the variable in my lattice struct, that means this variable is either a field or a formal parameter (as they are already initialized and I assume it couldn't be a variable that wasn't declared before)

File: src/ex3/VariableInitVisitor.java (hunk 4 above)
Comment (Elbargho) [reply src/ex3/VariableInitVisitor.java]: yes I guess, I added that check

**Pull request merged**